src/main/kotlin/com/openlattice/shuttle/IntegrationService.kt

```diff
@@ -0,0 +1,267 @@
+        private val missionParameters: MissionParameters,
+        private val idService: EntityKeyIdService,
+        private val entitySetManager: EntitySetManager,
+        private val reservationService: HazelcastAclKeyReservationService,
+        private val blackbox: Blackbox
+) {
+
+    private val integrations = hazelcastInstance.getMap<String, Integration>(HazelcastMap.INTEGRATIONS.name)
+    private val entitySets = hazelcastInstance.getMap<UUID, EntitySet>(HazelcastMap.ENTITY_SETS.name)
+    private val entityTypes = hazelcastInstance.getMap<UUID, EntityType>(HazelcastMap.ENTITY_TYPES.name)
+    private val propertyTypes = hazelcastInstance.getMap<UUID, PropertyType>(HazelcastMap.PROPERTY_TYPES.name)
+    private val integrationJobs = hazelcastInstance.getMap<UUID, IntegrationJob>(HazelcastMap.INTEGRATION_JOBS.name)
+    private val integrationQueue = hazelcastInstance.getQueue<Pair<UUID, String>>(HazelcastQueue.INTEGRATION_JOBS.name)
+    private val creds = missionParameters.auth.credentials
+    private val semaphore = Semaphore(nThreads)
+    private val executor = MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(nThreads))
+    private val statusPredicate = Predicates.or(
+            Predicates.equal(INTEGRATION_STATUS, IntegrationStatus.IN_PROGRESS),
+            Predicates.equal(INTEGRATION_STATUS, IntegrationStatus.QUEUED)
+    )
+
+    companion object {
+        @JvmStatic
+        fun init(blackbox: Blackbox, dataModelService: EdmManager) {
+            if (blackbox.enabled) {
+                logEntityType = dataModelService.getEntityType(FullQualifiedName(blackbox.entityTypeFqn))
+            }
+        }
+    }
+
+    init {
+        executor.execute {
+            semaphore.acquire()
+
+            //load any jobs that were in progress or queued
+            integrationJobs.entrySet(statusPredicate).forEach{
+                integrationQueue.put(it.key to it.value.integrationName)
+            }
+
+            while (true) {
+                if (semaphore.availablePermits() > 0) {
+                    loadCargo()
+                }
+            }
+        }
+    }
+
+    fun enqueueIntegrationJob(integrationName: String): UUID {
+        checkState(integrations.containsKey(integrationName), "Integration with name $integrationName does not exist")
+        val jobId = generateIntegrationJobId()
+        integrationQueue.put(Pair(jobId, integrationName))
+        integrationJobs[jobId] = IntegrationJob(integrationName, IntegrationStatus.QUEUED)
+        return jobId
+    }
+
+    private final fun loadCargo() {
+        val integrationJob = integrationQueue.take()
+        val jobId = integrationJob.first
+        val integration = integrations.getValue(integrationJob.second)
+        //val token = MissionControl.getIdToken(creds.getProperty("email"), creds.getProperty("password"))
+        val apiClient = ApiClient(integration.environment) { "testingtoken" }
```

Comment:
this seems like a TODO?

Reply:
Yes, this is the local auth0 stuff I was talking about